Drivers/PricePartDisplayDriver.cs

```diff
@@ -51,9 +54,37 @@ private Task BuildViewModel(PricePartViewModel model, PricePart part)
             model.PriceValue = part.Price.Value;
             model.PriceCurrency = part.Price.Currency == Currency.UnspecifiedCurrency ? _moneyService.DefaultCurrency.CurrencyIsoCode : part.Price.Currency.CurrencyIsoCode;
             model.PricePart = part;
-            model.Currencies = _moneyService.Currencies;
+
+            model.Currencies = GetCurrencySelectionList(part);
 
             return Task.CompletedTask;
         }
+
+        private IEnumerable<ICurrency> GetCurrencySelectionList(PricePart part)
+        {
+            IEnumerable<ICurrency> currencySelectionList;
+
+            switch (part.CurrencySelectionMode)
+            {
+                case CurrencySelectionModes.AllCurrencies:
```

Comment:
Since this is the default, you can remove that case entirely.

Reply:
Removed.